app/Http/Controllers/FacultyController.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Faculty;
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+
+class FacultyController extends Controller
+{
+    public function uuid()
+    {
+        return Str::uuid()->toString();
+    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create(Request $request)
+    {
+        $id = $this->uuid();
+        $facultyName = $request->get('faculty_name');
+        $facultyCode = $request->get('faculty_code');
+        $schoolID = $request->get('school_id');
+        $status = 'Active';
+        $faculty = new Faculty();
+        $faculty->id = $id;
+        $faculty->name = $facultyName;
+        $faculty->code = $facultyCode;
+        $faculty->school_id = $schoolID;
+        $faculty->status = $status;
+        
+
+        $checkSchool = Faculty::where(['name' => $facultyName])->first();
+        if(empty($checkSchool)){
+            if($faculty->save()){
+                return response()->json(['success' => true], 200);
+            }
+            return response()->json(['success' => false], 200);
+        }
+        return response()->json(['error' => 'This faculty already exist'], 200);
```

Comment:
@bynalab is there a reason we are returning status code 200 for an error response? Looks like a `409` will be ideal for this case, see https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists for details

Reply:
Thanks for the update.